kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -543,38 +543,78 @@ def _evm_base_sort_int(type_label: str) -> bool:
-    if type_label == 'int256':
-        return KEVM.range_sint(256, term)
-    if type_label == 'bytes':
-        return KEVM.range_uint(128, KEVM.size_bytes(term))
-    if type_label == 'string':
-        return TRUE
+    match type_label:
+        case 'address':
+            return KEVM.range_address(term)
+        case 'bool':
+            return KEVM.range_bool(term)
+        case 'bytes':
+            return KEVM.range_uint(128, KEVM.size_bytes(term))
+        case 'string':
+            return TRUE
+
+    predicate_functions = [
+        _range_predicate_uint,
+        _range_predicate_int,
+        _range_predicate_bytes,
+    ]
+
+    for f in predicate_functions:
+        (success, result) = f(term, type_label)
+        if success:
+            return result
 
     _LOGGER.info(f'Unknown range predicate for type: {type_label}')
     return None
 
 
 def _range_predicate_uint(term: KInner, type_label: str) -> tuple[bool, KInner | None]:
     if type_label.startswith('uint') and not type_label.endswith(']'):
-        width = int(type_label[4:])
+        if type_label == 'uint':
+            width = 256
+        else:
+            width = int(type_label[4:])
         if not (0 < width and width <= 256 and width % 8 == 0):
-            raise ValueError(f'Unsupported range predicate type: {type_label}')
+            raise ValueError(f'Unsupported range predicate uint<M> type: {type_label}')
         return (True, KEVM.range_uint(width, term))
     else:
         return (False, None)
 
 
+def _range_predicate_int(term: KInner, type_label: str) -> tuple[bool, KInner | None]:
+    if type_label.startswith('int') and not type_label.endswith(']'):
+        if type_label == 'int':
+            width = 256
+        else:
+            width = int(type_label[3:])
+        if not (0 < width and width <= 256 and width % 8 == 0):
+            raise ValueError(f'Unsupported range predicate int<M> type: {type_label}')
+        return (True, KEVM.range_sint(width, term))
+    else:
+        return (False, None)
+
+
+def _range_predicate_bytes(term: KInner, type_label: str) -> tuple[bool, KInner | None]:
+    if type_label.startswith('bytes') and not type_label.endswith(']'):
+        str_width = type_label[5:]
+        if str_width != '':
+            width = int(str_width)
+            if not (0 < width and width <= 32):
+                raise ValueError(f'Unsupported range predicate bytes<M> type: {type_label}')
+            return (True, KEVM.range_bytes(intToken(width), term))
+    return (False, None)
+
+
+# TODO write semantics that supports static arays
+def _range_predicate_static_array(term: KInner, type_label: str) -> tuple[bool, KInner | None]:
+    if type_label.endswith(']') and not type_label.endswith('[]'):
+        parts = type_label.rsplit('[', 1)
+        if len(parts) > 1:
+            number = int(parts[1].rstrip(']'))
+            return (True, KEVM.range_uint(((number + 1) * 256), intToken(32)))
+    return (False, None)
```

Comment:
Should we remove this or is it worth keeping it for later work on a better support of dynamic types, two-dimensional arrays, and structs ?

Reply:
If this code is not executed anywhere and tested, then I don't see a reason to keep it. I suggest adding it to an issue to support dynamic types as a starting point instead.

Reply:
Will do. Thanks !